src/tests/api-tests.ts

```diff
@@ -1912,6 +1912,255 @@ describe('api tests', () => {
+
+    // test without 0x-prefix
+    const searchResult2 = await supertest(api.server).get(
+      `/extended/v1/search/1234000000000000000000000000000000000000000000000000000000000000?include_metadata=true`
+    );
+    expect(searchResult2.status).toBe(200);
+    expect(searchResult2.type).toBe('application/json');
+    const expectedResp2 = {
+      found: true,
+      result: {
+        entity_id: '0x1234000000000000000000000000000000000000000000000000000000000000',
+        entity_type: 'block_hash',
+        block_data: {
+          canonical: true,
+          hash: '0x1234000000000000000000000000000000000000000000000000000000000000',
+          parent_block_hash: '0xff0011',
+          burn_block_time: 94869286,
+          height: 1,
+        },
+        metadata: blockMetadata,
+      },
+    };
+    expect(JSON.parse(searchResult2.text)).toEqual(expectedResp2);
+
+    // test whitespace
+    const searchResult3 = await supertest(api.server).get(
+      `/extended/v1/search/ 1234000000000000000000000000000000000000000000000000000000000000?include_metadata=true`
+    );
+    expect(searchResult3.status).toBe(200);
+    expect(searchResult3.type).toBe('application/json');
+    const expectedResp3 = {
+      found: true,
+      result: {
+        entity_id: '0x1234000000000000000000000000000000000000000000000000000000000000',
+        entity_type: 'block_hash',
+        block_data: {
+          canonical: true,
+          hash: '0x1234000000000000000000000000000000000000000000000000000000000000',
+          parent_block_hash: '0xff0011',
+          burn_block_time: 94869286,
+          height: 1,
+        },
+        metadata: blockMetadata,
+      },
+    };
+    expect(JSON.parse(searchResult3.text)).toEqual(expectedResp3);
+
+    // test mempool tx search
+    const searchResult4 = await supertest(api.server).get(
+      `/extended/v1/search/0x8912000000000000000000000000000000000000000000000000000000000000?include_metadata=1`
+    );
+    expect(searchResult4.status).toBe(200);
+    expect(searchResult4.type).toBe('application/json');
+    const expectedResp4 = {
+      found: true,
+      result: {
+        entity_id: '0x8912000000000000000000000000000000000000000000000000000000000000',
+        entity_type: 'mempool_tx_id',
+        tx_data: { tx_type: 'coinbase' },
+        metadata: {
+          anchor_mode: 'any',
+          coinbase_payload: {
+            data: '0x636f696e62617365206869',
+          },
+          fee_rate: '1234',
+          nonce: 0,
+          post_condition_mode: 'allow',
+          post_conditions: [],
+          receipt_time: 123456,
+          receipt_time_iso: '1970-01-02T10:17:36.000Z',
+          sender_address: 'sender-addr',
+          sponsored: false,
+          tx_id: '0x8912000000000000000000000000000000000000000000000000000000000000',
+          tx_status: 'success',
+          tx_type: 'coinbase',
+        },
+      },
+    };
+    expect(JSON.parse(searchResult4.text)).toEqual(expectedResp4);
+
```

Comment:
Can you add a test for a regular transaction? Looks like that's the only search result type missing a test.

Reply:
Sure, will do that.